test/asar.js

```diff
@@ -53,3 +53,23 @@ util.testSinglePlatform('asar test', (t, opts) => {
       return fs.stat(path.join(resourcesPath, 'app.asar.unpacked', 'dir_to_unpack'))
     }).then(stats => t.true(stats.isDirectory(), 'dir_to_unpack should exist under app.asar.unpacked subdirectory when opts.asar-unpack-dir is set dir_to_unpack'))
 })
+
+util.testSinglePlatform('prebuilt asar test', (t, opts) => {
+  opts.name = 'prebuiltAsarTest'
+  opts.dir = util.fixtureSubdir('asar-prebuilt')
+  opts.asar = {
+    'filename': path.join(opts.dir, 'app.asar')
+  }
+  let resourcesPath
+
+  return util.packageAndEnsureResourcesPath(t, opts)
+    .then(generatedResourcesPath => {
+      resourcesPath = generatedResourcesPath
+      return fs.stat(path.join(resourcesPath, 'app.asar'))
+    }).then(stats => {
+      t.true(stats.isFile(), 'app.asar should exist under the resources subdirectory when opts.asar.filename points to a prebuilt asar')
+      return fs.pathExists(path.join(resourcesPath, 'app'))
+    }).then(exists => {
+      return t.false(exists, 'app subdirectory should NOT exist when app.asar is built')
+    })
```

Comment:
This test should check that the copied asar is the same as the input asar. There is a utility function to assert this, I believe.